### DISCUSSION TITLE
Error in function createFiberFromTypeAndProps in ./node_modules/react-dom/cjs/react-dom.development.js:25058

### DISCUSSION THREAD

**Discussion by Trushar10** [gatsbyjs/gatsby#31534] [Help] [answered]
I am getting an error while creating a page for a single post. The reference [tutorial](https://www.youtube.com/watch?v=mdownVNA-ms&list=PLW0RabRDhwwzVNhlOgQQgw6HJzXdM1MnT&index=21) to learn gatsby. I have match the steps and it is okay. What is causing it? @hunterbecton

Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports. Check the render method of `singlePost`.

 25056 |           {
  25057 |             {
> 25058 |               throw Error( "Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: " + (type == null ? type : typeof type) + "." + info );
  25059 |             }
  25060 |           }
  25061 |         }

Here is the
```
import React from "react"
import {graphql} from "gatsby"
import {MDXRenderer} from "gatsby-plugin-mdx"
import {H1} from "../elements"
import {Container, Post, FeatureImage, Seo} from "../components"

const singlePost = ({data}) => {
    const featureImage =data.mdx.frontmatter.featureImage.childImageSharp.fixed

    const seoImage = data.mdx.frontmatter.featureImage.publicURL

    return (
        <Container>
        <Seo      
          title={data.mdx.frontmatter.title}
          image={seoImage}
          description= {data.mdx.frontmatter.excerpt}
        />
            <FeatureImage fixed={featureImage} />
            <Post>
                <H1 margin = "0 0 2rem 0">
                    {data.mdx.frontmatter.title}
                </H1>
                <MDXRenderer>{data.mdx.body}</MDXRenderer>
            </Post>
        </Container>
    )
}

export default singlePost

export const pageQuery = graphql `
    query SinglePostQuery($id: String!) {
        mdx(id: {eq: $id}) {
          body
          frontmatter {
            date
            excerpt
            slug
            title
            featureImage {
              childImageSharp {
                fixed {
                  ...GatsbyImageSharpFixed
                }
              }
            }
          }
        }
    }
`
```

**Comment by vladar** [comment]
One of your components is not exported: `H1`, `Container`, `Seo`. `FeatureImage` or `Post`. Can you share your repository so that we could check which one is actually missing?

**Reply by Trushar10** [reply]
@vladar https://github.com/Trushar10/hello-world.git

**Reply by Trushar10** [reply]
Any answer for the error? @vladar @hunterbecton

**Comment by Trushar10** [comment, marked as answer]
The component name was misspelled in the component file. The issue is resolved.
